tests/post.test.ts

```diff
@@ -0,0 +1,63 @@
+import { EventPayload } from '../src/EventPayload';
+import { EventAPIResponse } from '../src/EventAPIResponse';
+import post from '../src/post';
+const fetchMock = /** @type {FetchMockJest} */ (require('fetch-mock-jest'));
+
+describe('Test post util function', () => {
+
+    beforeEach(() => {
+        const navigator = { userAgent: 'Firefox', sendBeacon: () => { return true }};
+        Object.defineProperty(window, 'navigator', { value: navigator, writable: true});
+    });
+
+    const eventResponseA: EventAPIResponse = {
+        id: 'event-id-A'
+    }
+
+    const eventPayloadA: EventPayload = {
+        action: 'ADD_TO_CART',
+        locale: 'en_US',
+    };
+
+    const eventPayloadB: EventPayload = {
+        action: 'ADD_TO_CART',
+        locale: 'en_US',
+        browserAgent: {
+            browser: 'Firefox'
+        }
+    };
+
+    const url = 'https://dev.us.yextevents.com/accounts/me/events';
+
+    const optionsA = {method: 'POST', body: JSON.stringify(eventPayloadA), keepalive: true};
+
+    fetchMock.post(url, JSON.stringify(eventResponseA));
+
+    it('should send fetch request', async () => {
+        // forceFetch: true
+        await post(url, eventPayloadA, true);
+        expect(fetchMock.calls('https://dev.us.yextevents.com/accounts/me/events')[0][1]).toEqual(optionsA);
```

Comment:
I'm having trouble with mocking the response of the fetch call, so I'm only checking that fetch gets called correctly right now. Putting up a PR to be reviewed first while I look into it.

Reply:
Discussed with Merrick offline - mocking and then verifying the response is not really needed here as the main purpose of the tests are to check that it either calls fetch or sendBeacon.